static/moodytunes/js/playlists.js

```diff
@@ -1,28 +1,28 @@
 'use strict';
 
 (function IIFE() {
+    var generatePlaylistButton = document.getElementById('generate-playlist');
+
+    // Cache options for previous request, used for refreshing playlist on delete of vote
+    var lastEmotion,
+        lastGenre,
+        lastContext;
+
     function init() {
-        var generatePlaylistButton = document.getElementById('generate-playlist');
         generatePlaylistButton.addEventListener('click', getEmotionPlaylist);
     }
 
-    function deleteVote() {
-        var emotion = document.getElementById('id_emotion').value;
-        var song = this.dataset.song;
-        var context = document.getElementById('id_context').value || undefined;
-
-        document.MoodyTunesClient.deleteVote(song, emotion, context, function(data) {
-            // Disable buttons to prevent double votes for a track
-            var songContainer = document.getElementById('song-' + song);
-            songContainer.className += ' song-container-vote-delete';
-            var button = songContainer.querySelectorAll('button')[0];  // We've only got the one delete button
-            button.disabled = true;
-            button.className += ' vote-button-delete-chosen';
-
-            // Update analytics after delete has been processed
-            var genre = document.getElementById('id_genre').value || undefined;
-            document.MoodyTunesClient.getUserAnalytics(emotion, genre, context, displayAnalytics);
-        })
+    function deleteVote(evt) {
+        var confirmVal = confirm('Are you sure you want to remove this song from your playlist?');
```

Comment:
Let's try to find a more elegant way to confirm a delete request, preferably through a modal we can control

Reply:
After doing a POC of using a modal to confirm the delete request, I ran into a few problems with the click handler. It is possible for the user to click the delete button twice, sending two requests to delete the vote. We handle this on the backend, but it results in a poor user experience where they can incidentally send two requests to delete a vote.

The confirm workflow is assured to not have duplicate requests. Consider the tradeoff in appearance vs workflow

Reply:
We can use a modal for this purpose, it allows us greater control over the styling and appearance. It also allows us greater consistency between the different popups we use as they would look similiar